connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -141,13 +148,36 @@ public void stop() {
     }
 
     // divide topic-partitions among tasks
+    // since each mirrored topic has different traffic and number of partitions, to balance the load
+    // across all mirrormaker instances (workers), 'roundrobin' helps to evenly assign all
+    // topic-partition to the tasks, then the tasks are further distributed to workers by calling
+    // 'ConnectorUtils.groupPartitions()'. For example, 3 tasks to mirror 3 topics with 8, 2 and 2
+    // partitions respectively. 't1' denotes 'task 1', 't0p5' denotes 'topic 0, partition 5'
+    // t1 -> [t0p0, t0p3, t0p6, t1p1]
+    // t2 -> [t0p1, t0p4, t0p7, t2p0]
+    // t3 -> [t0p2, t0p5, t1p0, t2p1]
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
         if (!config.enabled() || knownTopicPartitions.isEmpty()) {
             return Collections.emptyList();
         }
         int numTasks = Math.min(maxTasks, knownTopicPartitions.size());
-        return ConnectorUtils.groupPartitions(knownTopicPartitions, numTasks).stream()
+        List<List<TopicPartition>> roundRobinByTask = new ArrayList<>(numTasks);
+        for (int i = 0; i < numTasks; i++) {
+            roundRobinByTask.add(new ArrayList<>());
+        }
+        int count = 0;
+        for (TopicPartition partition : knownTopicPartitions) {
+            int index = count % numTasks;
+            roundRobinByTask.get(index).add(partition);
+            count++;
+        }
+        List<TopicPartition> roundRobinAllTasks = new ArrayList<>();
+        for (List<TopicPartition> assignmentPerTask : roundRobinByTask) {
+            roundRobinAllTasks.addAll(assignmentPerTask);
+        }
+
+        return ConnectorUtils.groupPartitions(roundRobinAllTasks, numTasks).stream()
```

Comment:
Do we need to call `groupPartitions()`? Isn't `roundRobinAllTasks` already what we want?

Reply:
I think we need to call `groupPartitions()` to assign shuffled topic-paritition (saved in `roundRobinAllTasks`) to tasks [`0` - `n`] (e.g. `n` is 1 and 6 topic-partition, so size of `roundRobinAllTasks` is 6), where `task 0` takes the first 3 elements of `roundRobinAllTasks`, and `task 1` takes the rest of 3 elements

Reply:
Not sure I understand the difference here.
If I do not create `roundRobinAllTasks` and simply return:
```
return roundRobinByTask.stream()
            .map(config::taskConfigForTopicPartitions)
            .collect(Collectors.toList());
```
All the tests still pass. If you think building the `roundRobinAllTasks` and calling `groupPartitions()` is different, can you add a test that demonstrate it (ie make my code fail)?

Reply:
you are right - there is no difference. updated the PR

Reply:
@mimaison thanks for pointing this out. Please review it again